plugins/nurseshark/index.js

```diff
@@ -199,9 +199,22 @@ var nurseshark = {
     function createUploadIDsMap() {
       var uploads = _.where(data, {type: 'upload'});
       _.each(uploads, function(upload) {
-        uploadIDSources[upload.uploadId] = upload.source ||
-          ((upload.deviceManufacturers && Array.isArray(upload.deviceManufacturers) && upload.deviceManufacturers.length > 0) ?
-            upload.deviceManufacturers[0] : 'Unknown');
+        var source = 'Unknown';
+        if (upload.hasOwnProperty('source')) {
+          source = upload.source;
+        }
+        else if(upload.hasOwnProperty('deviceManufacturers') && Array.isArray(upload.deviceManufacturers) && upload.deviceManufacturers.length) {
+          // Uploader does not specify `source` for CareLink uploads, so they incorrectly get set to `Medtronic`, which should only be used for Medtronic Direct uploads.
+          // Check if manufacturer equals Medtronic, then check pumpSettings array for uploads with that upload ID and a source of `carelink`, then override appropriately.
+          if (upload.deviceManufacturers[0] === 'Medtronic' && _.where(data, {type: 'pumpSettings', uploadId: upload.uploadId, source: 'carelink'}).length) {
```

Comment:
`_.where` does a rather expensive deep comparison. `_.filter` works with the exact same syntax in this case (providing an object is a possible shorthand for `_.filter`), and I think it doesn't do the deep comparison (the lodash docs don't specify shallow or deep under `_.filter` but they do specify deep for `_.where`)...

So change to `_.filter` please?